model/proxy/alphav1/config/cfg.proto

```diff
@@ -0,0 +1,483 @@
+// Copyright 2016 IBM Corporation
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+syntax = "proto3";
+
+import "google/protobuf/any.proto";
+import "google/protobuf/wrappers.proto";
+
+package istio.proxy.v1alpha.config;
+option go_package = "config";
+
+// Glossary & concepts
+
+// Cluster - set of pods/VMs/containers where the service is running. The
+// members of a cluster share one or more common attributes. For e.g., all
+// pods in a cluster could share a common set of labels, or be running the
+// same version of the application binary.
+
+// Multiple clusters/service - In a continuous deployment scenario, for a
+// given service, there can be multiple clusters running potentially
+// different variants of the application binary. These variants are not
+// necessarily different API versions. They could be iterative changes to
+// the same service, deployed in different environments (prod, staging,
+// dev, etc.). Common scenarios where this occurs include A/B testing,
+// canary rollouts, etc. The choice of a particular cluster can be decided
+// based on various criterion (headers, url, etc.) and/or by weights
+// assigned to each cluster.
+
+// Cluster identifier: A name and one or more tags uniquely identify each
+// cluster in a service. Names and tags can be arbitrary strings. Parsing
+// of names and tags is specific to the platform (e.g., in kubernetes, name
+// could correspond to the service Name, while each tag string could be
+// parsed into a pod label (key:value))
+
+// Downstream service/cluster - client (browser or another service cluster)
+// calling the proxy/sidecar (typically to reach another service).
+
+// Upstream service - The remote service to which the proxy/sidecar is
+// talking to, on behalf of the Downstream service. There can be one or
+// more upstream clusters for a given upstream service (see the discussion
+// on versions above). The proxy would choose the upstream cluster based on
+// various routing rules.
+
+// Applications address only the upstream service using without knowledge
+// of individual upstream clusters. The actual choice of the cluster is
+// determined by the proxy, enabling the application code to decouple
+// itself from the evolution of dependent services.
+
+// Proxy level global configurations go here
+message ProxyConfig {
```

Comment:
Unless we decide to add more stuff into proxy config, such as envoy userid, etc.